cdap-app-fabric/src/main/java/io/cdap/cdap/app/runtime/ProgramOptions.java

```diff
@@ -70,7 +72,8 @@ static ProgramOptions fromNotification(Notification notification, Gson gson) {
 
     boolean debug = Boolean.parseBoolean(debugString);
     Map<String, String> userArguments = userArgumentsString == null ?
-      Collections.emptyMap() : gson.fromJson(userArgumentsString, stringStringMap);
+      new HashMap<>() : gson.fromJson(userArgumentsString, stringStringMap);
+    userArguments.putIfAbsent(ProgramOptionConstants.RUN_ID_MACRO, programRunId.getRun());
```

Comment:
We shouldn't mess with user arguments. Seems like it should be a system arg instead?

Reply:
Kept it as a user arg in case user keeps an argument by the same name. In that case, that value would be picked. Do we change it to sys arg?